components/date-time/style.scss

```diff
@@ -122,3 +121,42 @@ $datepicker__border-color: $light-gray-500;
 .react-datepicker__close-icon::after {
 	background-color: theme( primary );
 }
+
+.date-time-form {
+	display: flex;
+	justify-content: center;
+	align-items: right;
+	margin-top: 10px;
+
+	.input-container {
+		padding: 0 4px;
+		flex: 0 1 auto;
+		order: 1;
+		&.day {
+			order: 0;
+		}
+
+		input[name=day] {
+			width: 40px;
+			-moz-appearance:textfield;
```

Comment:
I would have thought our build tools would do browser-prefixes as-needed.

Are these here as browser-specific styles (I recall `appearance` being different in Firefox from my days working on the Add-ons website, but I forget now). If they're browser-specific that's fine, but a comment explaining what they're for (and why they're `-moz`-only) would be handy 😄

(Also there should be a space after the colon but that's just a style nitpick.)

Reply:
Looks like some junk code from when I was working with number inputs. Well spotted!

Reply:
And regarding the design of the date picker widget itself, I feel like it's outside of the scope of the ticket, but I'll be happy to make a separate PR for that.